### DISCUSSION TITLE
Why does next-auth only expose rest apis and not direct methods for server-side code ?

### DISCUSSION THREAD

**Discussion by slevy85** [nextauthjs/next-auth#1412] [Ideas]
I read in the Next documentation that calling custom apis from getServerSideProps is not recommended : 

> Note: You should not use fetch() to call an API route in getServerSideProps. Instead, directly import the logic used inside your API route. You may need to slightly refactor your code for this approach.
> 
> Fetching from an external API is fine!

This is normal because server-side we already have access to everything a custom api would have access to, so we can call the functions directly without making some extra fetches on localhost.

I supposed that the issue is that the configuration of next-auth is in the file /pages/api/auth/[...nextauth].js, so the configuration becomes available only for the api routes. 
My suggestion would be to have for example a config file at root `next-auth.config.js` or a file that exposes an instance maybe `lib/next-auth.js`, and then we could use this instance directly : 
```
import NextAuth from 'lib/next-auth'

function getServerSideProps(context) {
   const session = NextAuth.getSessionWithoutFetching(context)
}
```

I am not sure maybe I am missing something.
What do you think ?

**Comment by balazsorban44** [comment]
This has been answered before. See https://github.com/nextauthjs/next-auth/issues/947#issuecomment-762150858

We have been playing with the thought of a config file of sorts, but we have to think more about how to do it right.

**Reply by slevy85** [reply]
Thank you I searched for this but I didn't find.
